grant-project/grant-monorepo
# Functioning proposal teams (pt 1 - the form)

### What This Does

This is the first of a few parts for getting proposal teams working. This adds a new step to the create flow for putting together your team.

### Things to Improve

* Pulling in existing user data needs to be hooked up if an email or eth address is entered that we know of. Might want to rethink how adding an existing user works.
* Validation works, but could definitely use improvements.

### Things Still Needed

* Proposal views (card, page) don't reflect the team yet, still uses trustees.
* Photos selection doesn't exist right now, it just cycles through random photos. This will require more discussion.
* Once auth is implemented, first team member should autofill as the auth'd user.
* Social accounts will need some form of validation / uniqueness.

### Steps to Test

* Create a new proposal
* Fill out a team, attempt to enter invalid data and make sure you can't
* Go to the review step, make sure your team is accurately shown

## Screenshots

### Team Step

<img width="875" alt="screen shot 2018-09-27 at 10 48 34 am" src="https://user-images.githubusercontent.com/649992/46154246-b8459c80-c242-11e8-8db6-169e85546244.png">

### Team Member Edit

<img width="647" alt="screen shot 2018-09-27 at 10 47 52 am" src="https://user-images.githubusercontent.com/649992/46154217-a82dbd00-c242-11e8-933c-f2ba616c1180.png">
